modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfBottomSheetPage.kt

```diff
@@ -0,0 +1,175 @@
+    trimmedShelf?.let { (shelfItems, episode) ->
+        shelfViewModel.setData(shelfItems, episode)
+        Content(
+            onEditButtonClick = {
+                shelfViewModel.onEditButtonClick()
+                onEditButtonClick()
+            },
+            mediaRouteButton = {
+                MediaRouteButton(
+                    performClick = performMediaRouteButtonClick,
+                    onMediaRouteButtonClick = {
+                        shelfViewModel.onMediaRouteButtonClick()
+                    },
+                )
+            },
+        ) {
+            MenuShelfItems(
+                shelfViewModel = shelfViewModel,
+                onClick = { item, enabled ->
+                    coroutine.launch {
+                        if (item == ShelfItem.Cast) {
+                            performMediaRouteButtonClick = true
+                            delay(100) // allow perform action to complete before dismissing the bottom sheet
+                        }
+                        onShelfItemClick(item, enabled)
+                    }
+                },
+            )
+        }
+    }
+}
+
+@Composable
+private fun Content(
+    onEditButtonClick: () -> Unit,
+    mediaRouteButton: @Composable () -> Unit,
+    menuShelfItems: @Composable () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth(),
+    ) {
+        Spacer(Modifier.height(8.dp))
+
+        Pill(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .background(MaterialTheme.theme.colors.playerContrast01),
+        )
+
+        Spacer(Modifier.height(8.dp))
+
+        mediaRouteButton()
+
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp),
+        ) {
+            TextH30(
+                text = stringResource(LR.string.player_more_actions),
+                color = MaterialTheme.theme.colors.playerContrast01,
+            )
+
+            IconButton(
+                onClick = onEditButtonClick,
+            ) {
+                Icon(
+                    painter = painterResource(IR.drawable.ic_edit),
+                    contentDescription = stringResource(LR.string.edit),
+                    tint = MaterialTheme.theme.colors.playerContrast01,
+                )
+            }
+        }
+        menuShelfItems()
+    }
+}
+
+@Composable
+private fun MediaRouteButton(
+    performClick: Boolean,
+    onMediaRouteButtonClick: () -> Unit,
+) {
+    AndroidView(
+        factory = { context ->
+            MediaRouteButton(context).apply {
+                visibility = View.GONE
+                setOnClickListener {
+                    onMediaRouteButtonClick()
+                }
+                CastButtonFactory.setUpMediaRouteButton(context, this)
```

Comment:
Make sure to test cast shelf item action properly as it depends on Compose <-> Android View interoperability.